pmd-core/src/main/java/net/sourceforge/pmd/renderers/HTMLRenderer.java

```diff
@@ -134,6 +148,20 @@ private void glomRuleViolations(Writer writer, Iterator<RuleViolation> violation
         }
     }
 
+    private String renderFileName(String filename, int beginLine) {
+        return maybeWrap(StringEscapeUtils.escapeHtml4(determineFileName(filename)),
+                linePrefix == null || beginLine < 0 ? "" : linePrefix + beginLine);
```

Comment:
The control flow confused me so I added another test. What happens when the link prefix is set but not the line prefix? Currently the link is rendered, but there's nothing behind the `#`. I think this might be useful behavior (no line prefix != empty line prefix), let me know if this is expected and I'll merge the PR

Reply:
I think we should preserve whatever current behavior it is (not sure, what exactly is correct/expected). But "no line prefix!=empty line prefix" sounds reasonable.

The use cases are:
* noLinePrefix - then, as it is implemented like now, we don't output the line number in the link
* LinePrefix is an empty string - then we should output "#4711". Some systems might use directly the linenumbers as target
* LinePrefix is any non-empty string - like github uses "L". Then we should output "#L4711".

I'll double check, that we have a test case for the empty lineprefix and merge afterwards.

Reply:
SGTM. Side note: this will make it so, that the property wil need to be changed to `Optional<String>` on 7.0 because the new property framework explicitly forbids null property values. So `Optional.empty()` would mean "no line prefix" instead of null